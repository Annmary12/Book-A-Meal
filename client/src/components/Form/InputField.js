import React from 'react';
import PropTypes from 'prop-types';
import styles from '../../../assets/scss/form.scss';

const InputField = ({
  values,
  touched,
  errors,
  handleChange,
  handleBlur,
  name,
  label,
  type
}) => (
  <label className={styles.label} htmlFor={name}>
    {label}:
    <input
      className={
        errors[name] && touched[name]
          ? styles.inputError : styles.input}
      type={type}
      name={name}
      onChange={handleChange}
      onBlur={handleBlur}
      value={values[name]}
    />
    {
      errors[name] &&
      touched[name] &&
      <div className={styles.errorMessage}>{errors[name]}</div>
    }
  </label>
);

InputField.propTypes = {
  values: PropTypes.objectOf(PropTypes.string).isRequired,
  touched: PropTypes.objectOf(PropTypes.bool).isRequired,
  errors: PropTypes.objectOf(PropTypes.string).isRequired,
  handleChange: PropTypes.func.isRequired,
  handleBlur: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.string
};

InputField.defaultProps = {
  type: 'text'
};

export default InputField;
