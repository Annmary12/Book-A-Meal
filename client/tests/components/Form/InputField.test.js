import React from 'react';
import InputField from '../../../src/components/Form/InputField.js';

describe("InputField component", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <InputField
        errors={{}}
        touched={{}}
        values={{}}
      />
    );

    expect(wrapper).toMatchSnapshot();
  })
})