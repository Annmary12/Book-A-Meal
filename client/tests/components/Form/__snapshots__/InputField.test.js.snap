// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField component should match snapshot 1`] = `
<label
  className="label"
>
  :
  <input
    className="input"
    type="text"
  />
</label>
`;
